### PR TITLE
docs: restructure and add 'Tinkering' section

### PR DESCRIPTION
**Pull request by Stadicus**
Because:
* The technical documentation should be up to date.
* Additional information and TODO markers help add additional details.
* Adding the 'Tinkering' section for beta testers, so they have a first
  explanation of the steps we are using in development regularly, but
  could be hard to figure out without guidance.

This commit:
* adds a 'Tinkering' section to the technical documentation
* restructures the technical docs
* updates existing information and images

Note:
* There are now quite a few empty sections that need to be addressed
  by various people. @Stadicus creates follow-up issues.

### CODE REVIEW

File: docs/customapps.md [hunk 1]
```diff
@@ -6,3 +6,5 @@ has_children: true
 permalink: /customapps
 ---
 # BitBoxBase: Custom applications
+
+This section outlines the use and configuration of custom applications we built for the BitBoxBase.
```
Comment (TheCharlatan) [inline docs/customapps.md]: More to come I guess, or did you leave this blank by accident?

Comment (TheCharlatan) [reply docs/customapps.md]: Ah, it's just for navigation, never mind.

Comment (Stadicus) [reply docs/customapps.md]: Yes, a table of content for subpages is generated automatically.

File: docs/tinkering/ssh.md [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+  Currently, the keys need to be added manually, either by logging in locally or after login in with a password (see next option). We plan to allow users to add SSH keys from the BitBoxApp.
+* **Password login**: this authentication method is not secure and should not be enabled for longer periods on a production device.
+
+  It can be enabled in the BitBoxApp node management under "Advanced options". Alternatively, you can run `sudo bbb-config.sh enable sshpwlogin` directly on the command line. After enabling, you can log in with the user `base` using the password set in the Setup wizard.
+* **Root login**: SSH access for the `root` user is disabled by default. For development, it can be enabled from the command line, e.g. to copy updated scripts directly into system folders that require root access. On the BitBoxBase, logged in with user `base`, run `sudo bbb-config.sh enable rootlogin`.
+
+If you build the BitBoxBase image yourself, you can configure the options `BASE_LOGINPW` (initial login password, overwritten by the Setup Wizard) and `BASE_SSH_PASSWORD_LOGIN` in [build.conf](https://github.com/digitalbitbox/bitbox-base/blob/master/armbian/base/build.conf).
```
Comment (0xB10C) [inline docs/tinkering/ssh.md]: > initial login password, overwritten by the Setup Wizard

Is that implemented already?

File: docs/update/update-method.md [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+We are working towards the goal to build the BitBoxBase as an appliance with 'firmware', not as a small Linux server.
+This is why we decided to implement a full disk image update process, using the [Mender](https://mender.io/) open-source solution.
+
+This solution has the following features:
+
+* **enterprise-grade**: despite being open-source and free to use, Mender is a professional solution with its own Deployment Management server and a robust client update daemon.
+* **full disk image**: the whole operating system including all applications are updated together, resulting in a clearly defined state of the system.
+* **atomic**: the update process is atomic, as it either succeeds in full (including custom application tests), or not at all. In case of failure, a full rollback is performed.
+* **on-demand**: the BitBox App will prompt you when an update is available, but you're not forced to update.
```
Comment (0xB10C) [inline docs/update/update-method.md]:
```suggestion
* **on-demand**: the BitBoxApp will prompt you when an update is available, but you're not forced to update.
```

File: docs/update/update-method.md [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+* **secure**: all updates are provided over a secure TLS communication connection and are cryptographically signed by Shift Cryptosecurity. By default, no unsigned updates are accepted by the device.
+* **efficient**: update images are compressed and are streamed directly to the device. As of today, a full update (operating system and all applications) is ~300 MB in size.
+* **minimal downtime**: the update is performed in the background, while applications continue to run. Only a single reboot is necessary to start the updated configuration.
+* **persistent data**: the eMMC contains multiple partitions, one of which is used to store persistent data like the device configuration.
+* **custom Base image**: more in the #reckless category, updating custom-built Base image can be enabled through the BitBox App.
```
Comment (0xB10C) [inline docs/update/update-method.md]:
```suggestion
* **custom Base image**: more in the #reckless category, updating custom-built Base image can be enabled through the BitBoxApp.
```

Comment (0xB10C) [reply docs/update/update-method.md]: Might be worth grepping to check if it's changed everywhere else.

File: docs/tinkering/ssh.md (hunk 2 above)
Comment (Stadicus) [reply docs/tinkering/ssh.md]: Yes, setting the management password from the App also updates the login password for users `base` and `root` (which are disabled for SSH login by default).

File: docs/update/update-method.md (hunk 3 above)
Comment (Stadicus) [reply docs/update/update-method.md]: Good find, replaced it globally.

**Comment by Stadicus** [comment]
Addressed feedback in https://github.com/digitalbitbox/bitbox-base/pull/253/commits/694cd12d56220c9b60c6f35c20c113dfd230a667

**Pull request merged**